Mathlib/Tactic/ToAdditive/GuessName.lean

```
@@ -59,183 +59,167 @@ partial def _root_.String.splitCase (s : String) (i₀ : Pos := 0) (r : List Str
-  | "le" :: "_" :: "zero" :: s        => "nonpos" :: fixAbbreviation s
-  | "LT" :: "Zero" :: s               => "Neg" :: fixAbbreviation s
-  | "lt" :: "_" :: "zero" :: s        => "neg" :: fixAbbreviation s
-  | "Add" :: "Single" :: s            => "Single" :: fixAbbreviation s
-  | "add" :: "Single" :: s            => "single" :: fixAbbreviation s
-  | "add" :: "_" :: "single" :: s     => "single" :: fixAbbreviation s
-  | "Add" :: "Support" :: s           => "Support" :: fixAbbreviation s
-  | "add" :: "Support" :: s           => "support" :: fixAbbreviation s
-  | "add" :: "_" :: "support" :: s    => "support" :: fixAbbreviation s
-  | "Add" :: "TSupport" :: s          => "TSupport" :: fixAbbreviation s
-  | "add" :: "TSupport" :: s          => "tsupport" :: fixAbbreviation s
-  | "add" :: "_" :: "tsupport" :: s   => "tsupport" :: fixAbbreviation s
-  | "Add" :: "Indicator" :: s         => "Indicator" :: fixAbbreviation s
-  | "add" :: "Indicator" :: s         => "indicator" :: fixAbbreviation s
-  | "add" :: "_" :: "indicator" :: s  => "indicator" :: fixAbbreviation s
-  | "is" :: "Even" :: s             => "even" :: fixAbbreviation s
-  | "Is" :: "Even" :: s             => "Even" :: fixAbbreviation s
+def abbreviationDict : String → Option String
+  | "isCancelAdd"      => "IsCancelAdd"
+  | "isLeftCancelAdd"  => "IsLeftCancelAdd"
+  | "isRightCancelAdd" => "IsRightCancelAdd"
+  | "cancelAdd"        => "AddCancel"
+  | "leftCancelAdd"    => "AddLeftCancel"
+  | "rightCancelAdd"   => "AddRightCancel"
+  | "cancelCommAdd"    => "AddCancelComm"
+  | "commAdd"          => "AddComm"
+  | "zero_le"          => "Nonneg"
+  | "zeroLE"           => "Nonneg"
+  | "zero_lt"          => "Pos"
+  | "zeroLT"           => "Pos"
+  | "lezero"           => "Nonpos"
+  | "le_zero"          => "Nonpos"
+  | "ltzero"           => "Neg"
+  | "lt_zero"          => "neg"
+  | "addSingle"        => "Single"
+  | "add_single"       => "Single"
```

Are you sure that `mul_single` needs to be turned into `single`? I'd think this only applies to `mulSingle`?

---

I got some confusing behaviour when working on `MonoidAlgebra` with lemmas that talked about multiplying with `single`. (Although in that case the additive name was also supposed to be about `mul` 😅)

---

The confusing behaviour was that it generated some stupid additive name, and this stupid additive name actually was an existing lemma. So then it gave a confusing error about the `to_additive` tag being wrong. Maybe to avoid that, it should just throw an error when not using `existing`, and if the constant already exists.